File: backend/src/models/Unit.ts
// Unit types
export enum UnitType {
  APARTMENT = 'apartment',
  HOUSE = 'house',
  VILLA = 'villa',
  COMMERCIAL = 'commercial',
  OFFICE = 'office',
  SHOP = 'shop',
  STUDIO = 'studio',
  ROOM = 'room'
}

// Unit status
export enum UnitStatus {
  AVAILABLE = 'available',
  OCCUPIED = 'occupied',
  UNDER_MAINTENANCE = 'under_maintenance',
  VACANT = 'vacant'
}

export interface Unit {
  id: string; // UUID
  propertyId: string; // UUID reference to properties table
  unitNumber: string; // e.g., "101", "Villa-A", "Office-1"
  unitName?: string; // Optional display name
  description?: string;
  unitType: UnitType;
  status: UnitStatus;

  // Unit specifications
  floor?: number;
  area: number; // in sq ft
  bedrooms?: number;
  bathrooms?: number;
  balconies?: number;
  furnished: boolean;
  maxOccupants?: number;

  // Unit amenities and features
  unitAmenities: string[]; // ['wifi', 'parking', 'ac', 'tv', etc.]
  unitPhotos: string[]; // array of photo URLs

  // Financial details (per unit)
  monthlyRent: number;
  securityDeposit: number;
  maintenanceCharges?: number;

  // Metadata
  createdAt: Date;
  updatedAt: Date;
}

export interface UnitInput {
  propertyId: string;
  unitNumber: string;
  unitName?: string;
  description?: string;
  unitType: UnitType;
  status?: UnitStatus;

  // Unit specifications
  floor?: number;
  area: number;
  bedrooms?: number;
  bathrooms?: number;
  balconies?: number;
  furnished?: boolean;
  maxOccupants?: number;

  // Unit amenities and features
  unitAmenities?: string[];
  unitPhotos?: string[];

  // Financial details
  monthlyRent: number;
  securityDeposit: number;
  maintenanceCharges?: number;
}

// Unit-Tenant relationship for shared housing support
export interface UnitTenant {
  id: string; // UUID
  unitId: string; // UUID reference to units table
  tenantId: string; // UUID reference to tenants table
  isPrimaryTenant: boolean;

  // Occupancy details
  moveInDate?: Date;
  moveOutDate?: Date;

  // Financial sharing
  monthlyRentShare: number; // Amount this tenant pays
  securityDepositShare: number; // Amount this tenant contributes

  status: 'active' | 'inactive' | 'evicted';

  // Metadata
  createdAt: Date;
  updatedAt: Date;
}

export interface UnitTenantInput {
  unitId: string;
  tenantId: string;
  isPrimaryTenant?: boolean;
  moveInDate?: Date;
  moveOutDate?: Date;
  monthlyRentShare: number;
  securityDepositShare: number;
  status?: 'active' | 'inactive' | 'evicted';
}

File: backend-dot-net-core/src/MyApp.Models/Unit.cs
using System;

namespace MyApp.Models;

public class Unit
{
    public Guid Id { get; set; }
    public Guid PropertyId { get; set; }

    // Business identifiers
    public string UnitNumber { get; set; } = string.Empty; // unit_number (required)
    // Friendly name/label
    public string Name { get; set; } = string.Empty; // unit_name

    public string? Description { get; set; }
    public string? UnitType { get; set; }

    public int? Floor { get; set; }
    public decimal? Area { get; set; }
    public int? Bedrooms { get; set; }
    public int? Bathrooms { get; set; }
    public int? Balconies { get; set; }
    public bool? Furnished { get; set; } = false;
    public int? MaxOccupants { get; set; }

    // JSON blobs
    public string? UnitAmenities { get; set; }
    public string? UnitPhotos { get; set; }

    // Financials
    public decimal? MonthlyRent { get; set; }
    public decimal? SecurityDeposit { get; set; }
    public decimal? MaintenanceCharges { get; set; }

    public string? Status { get; set; } = "available";

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}